src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -791,8 +797,9 @@ void BlockchainSQLite::blockchain_detached(
         prepared_exec(
                 "INSERT INTO delayed_payments ({delayed_fields})"
                 " SELECT {delayed_fields} FROM delayed_payments_{suffix}"
-                " WHERE ? BETWEEN height AND payout_height"_format(
+                " WHERE ? >= height AND ? < payout_height"_format(
                         "delayed_fields"_a = DELAYED_PAYMENTS_COLS, "suffix"_a = suffix),
+                static_cast<int64_t>(new_height),
```

Comment:
SQLite indexed binding (?NNN) avoids needing to bind the same value twice:
```suggestion
                " WHERE ?1 >= height AND ?1 < payout_height"_format(
                        "delayed_fields"_a = DELAYED_PAYMENTS_COLS, "suffix"_a = suffix),
```